### PR TITLE
Add markdown support for description field

### PR DESCRIPTION
**Pull request by alankbi**
### List of Changes:

* Add markdown support for description field
* Change database description field type to TEXT (for unlimited length)
* Change game page problem display to properly show multi-line and markdown text

### Notes:

* This PR uses this markdown library: https://github.com/outline/rich-markdown-editor
  * It's very fancy and high-performing, though the tradeoff might be lower flexibility/reliance on a company product

### Screencast and Images:

[Screencast of markdown editing and viewing behavior](https://www.loom.com/share/7e3300d04948409d8e25dbf7240d7893)

Game page with markdown description:

Game page with markdown description

Editing display for markdown:

Editing page for markdown

### CODE REVIEW

**Comment by alankbi** [comment]
[Change description field to markdown](https://trello.com/c/f3Hnm6EC/264-change-description-field-to-markdown)

**Review by zpChris — CHANGES_REQUESTED** [review]
Requesting changes, as there is a slight issue (that we noticed on the call today) with the Markdown going past the bottom of the Game Page section. Additionally, we may want to move the placement of the "socket not connected error" - see the [screencast](https://www.loom.com/share/6c40bd28e3a44593925db1de490f9f32).

As for your note:
> It's very fancy and high-performing, though the tradeoff might be lower flexibility/reliance on a company product

I think that we definitely want to use an already-implemented Markdown editor in our app - there are lots of reliable sources out there, so we could switch it relatively easily if need-be, and I think the fanciness and high-performance is just a plus 😁

File: frontend/src/views/Game.tsx [hunk 1]
```diff
@@ -336,7 +342,11 @@ function GamePage() {
           {/* Problem title/description panel */}
           <Panel>
             <ProblemHeaderText>{problems[0]?.name}</ProblemHeaderText>
-            <Text>{problems[0]?.description}</Text>
+            <StyledMarkdownEditor
+              defaultValue={problems[0]?.description}
+              onChange={() => ''}
```
[the remark this answers is not in the document]
Comment (alankbi) [reply frontend/src/views/Game.tsx:347]: Without it I get a TypeScript "no overload matches this call" error, so I think it requires you to have something specified (unless there's a way around it if you know)

File: src/main/java/com/rocketden/main/model/problem/Problem.java [hunk 2]
```diff
@@ -30,6 +31,7 @@
     private String problemId = UUID.randomUUID().toString();
 
     private String name;
+    @Column(columnDefinition = "TEXT")
```
[the remark this answers is not in the document]
Comment (alankbi) [reply src/main/java/com/rocketden/main/model/problem/Problem.java:34]: The default I think is 255 characters (VARCHAR(255) or something like that). For reference, this is what ~230 characters look like: 

> Find the ASCII character that occurs most frequently in the string. If there is a tie, return the first character that appears in the string. Return an empty character if the string is empty. You can assume that the string is not null. 

Do you think any of our other string fields (for any database object) should be updated to be longer than 255?

**Comment by alankbi** [comment]
Pushed a change that should fix the overflow error and also moved the error text to below the title/above the description - lmk your thoughts on that

File: src/main/java/com/rocketden/main/model/problem/Problem.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (zpChris) [reply src/main/java/com/rocketden/main/model/problem/Problem.java:34]: We may want to later, but I don't think there's any need to now - we can change it when necessary in another PR.

**Review by zpChris — APPROVED** [review]
LGTM! Thanks for making the overflow panel change, looks great 👍

**Pull request merged**